### PR TITLE
Add icclim python package

### PR DESCRIPTION
**Pull request by bzah**
This PR adds icclim recipe.
icclim is a python package to compute ECA&D climate indices. 

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/icclim/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    - pip
+    - pytest
+
+about:
+  summary: 'icclim (Index Calculation CLIMate) is a python library for climate indices calculation.'
+  home: https://github.com/cerfacs-globc/icclim
+  doc_url: https://icclim.readthedocs.io/
+  dev_url: https://github.com/cerfacs-globc/icclim
+  license: Apache-2.0
+  license_file: LICENCE
```
Comment (zklaus) [inline recipes/icclim/meta.yaml:54]: I think you need to change the packaging of the tarball slightly. The file `LICENSE` (sic, i.e. with an S to conform to American English) should contain the actual license, i.e. be a copy of https://www.apache.org/licenses/LICENSE-2.0.txt. What you have in `LICENCE` right now might be appropriately placed in `NOTICE`. What you have in `NOTICE` can go to `README` or anything else. More info about those files from the perspective of the Apache Foundation can be found at https://infra.apache.org/licensing-howto.html

There may be some room for interpretation, but one thing is clear: The conda-forge distribution *must* contain the actual license, not just a pointer to it.

**Review by zklaus — CHANGES_REQUESTED** [review]
I left an initial review. The first goal is to get everything to build correctly; then we can do another round.

Nice to see icclim added to conda-forge!

File: recipes/icclim/meta.yaml (hunk 1 above)
Comment (bzah) [reply recipes/icclim/meta.yaml:54]: This is very informative, thank you !

File: recipes/icclim/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    - python >=3.8
+    - xclim >=0.33
+    - numpy >=1.21.1
+    - xarray >=0.19
+    - dask[array] >=2021.10.0
+    - netCDF4 >=1.5.7
+    - cftime >=1.5.0
+    - pyyaml >=6.0
+    - cftime >=1.4.1
+    - setuptools-git >=1.2
```
[the remark this answers is not in the document]
Comment (bzah) [reply recipes/icclim/meta.yaml]: Indeed it's useless. It will be removed in 5.0.1 in favor of a `MANIFEST.in` file.

**Comment by pagecp** [comment]
I'm a maintainer.

**Comment by zklaus** [comment]
@conda-forge-admin, please restart ci

**Comment by zklaus** [comment]
Nice progress, @bzah! I was gonna write that we can ignore the windows fail since this is a `noarch`, but looking at the log, it seems to uncover a bit of a bug that makes icclim unnecessarily incompatible with windows.

You should *not* change the time zone to GMT for the logger, but rather change the used `Formatter` to use GMT if that is what's desired. This automatically avoids the issue of `time.tzset` not being available on anything but Unix.
Have a look at https://docs.python.org/3/library/logging.html#logging.Formatter.formatTime and https://stackoverflow.com/questions/6321160/how-to-set-timestamps-on-gmt-utc-on-python-logging, and consider changing https://github.com/cerfacs-globc/icclim/blob/05d49bf4177b6fc0eda186153e2994b64c97cfea/icclim/icclim_logger.py#L53-L55 accordingly.

I also have another comment regarding `NOTICE`. This file now assigns copyright to the Apache Foundation. I don't think that's correct. You might want to consider changing that copyright and getting rid of the square brackets in the file as well.

**Comment by bzah** [comment]
Thanks a lot @zklaus !
I will try to fix these in a 5.0.2 asap.

**Comment by zklaus** [comment]
@conda-forge-admin, please restart ci

**Comment by zklaus** [comment]
Looking good. Finally, let's have a look at those pins. You are pinning every dependency with a minimal version. Do these need to be quite so tight? For example, have a look at [the global pins](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml). Does Icclim really not work with Numpy 1.19?

**Comment by bzah** [comment]
> Looking good. Finally, let's have a look at those pins. You are pinning every dependency with a minimal version. Do these need to be quite so tight? For example, have a look at [the global pins](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml). Does Icclim really not work with Numpy 1.19 ?

The only mandatory pins are python 3.8 and Xclim 0.33.
I think it should be fine with older versions for other dependencies.

**Comment by zklaus** [comment]
In that case, I would suggest removing the other pins. This will give greater flexibility to the dependency resolution.

**Comment by zklaus** [comment]
Almost there! I suggest ordering the dependencies alphabetically. This will help spot the duplicate `cftime`.

**Comment by zklaus** [comment]
I restarted the ci because the previous run on windows failed due to bad luck: Right in the middle of it a new version of the `pillow` dependency came through, but some form of race condition with the CDN prevented its proper installation. Now everything seems to have worked.

**Comment by zklaus** [comment]
Just to be sure, @bzah, icclim does not come with a cli, right?

Well done, I approve this PR and the addition of icclim to conda-forge. @conda-forge/staged-recipes, can someone have a look and merge?

File: recipes/icclim/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+test:
+  requires:
+    - pip
+    - pytest
+  imports:
+    - icclim
+    - icclim.models
+  commands:
+    - pip check
```
Comment (carterbox) [inline recipes/icclim/meta.yaml:43]:
```suggestion
test:
  requires:
    - pip
  imports:
    - icclim
    - icclim.models
  commands:
    - pip check
```
Pytest is unused, right? Your tests are only imports and dependency checking.

**Review by carterbox — APPROVED** [review]
Other than possibly extra test dependency, LGTM!

**Comment by bzah** [comment]
> I restarted the ci because the previous run on windows failed due to bad luck: Right in the middle of it a new version of the `pillow` dependency came through, but some form of race condition with the CDN prevented its proper installation. Now everything seems to have worked.

Oh I was wondering what happened with this windows build !
As for the CLI, we don't have one on icclim.

Thanks a lot for your patient review Klaus !

**Pull request merged**